### PR TITLE
Rename `mozilla-ion` to `mozilla-rally`

### PR DESCRIPTION
**Pull request by Dexterp37**
This contributes to mozilla-rally/internal#18

Note that this will require cutting a new release of the Core Add-on and a new release of the partner support library. Then, we'll need a new release of the study example, too.

Checklist for reviewer:

- [ ] The description should reference a bug or github issue, if relevant.
- [ ] There must be a [`CHANGELOG.md`](./CHANGELOG.md) entry for any non-test change.
- [ ] Any change to the NPM commands must be carefully reviewed to make sure it won't break the Add-ons pipeline.
- [ ] Any version increase must follow the [release process](./docs/RELEASE_PROCESS.md).

### CODE REVIEW

**Comment by rhelmer** [comment]
>     * [ ]  Any change to the NPM commands must be carefully reviewed to make sure it won't break the Add-ons pipeline.

Speaking of which, https://github.com/mozilla-extensions/ion-core-addon/ needs to be renamed too - @Dexterp37 do you mind filing a followup issue or PR for this? Sorry if it's already covered and I missed it!

File: bin/prepare-release.sh [hunk 1]
```diff
@@ -88,7 +88,7 @@ run rm "${WORKSPACE_ROOT}/${FILE}.bak"
 
 FILE=manifest.json
 run $SED -i.bak -E \
-    -e "s/^\"version\": \"[0-9a-z.-]+\"/\"version\": \"${NEW_VERSION}\"/" \
+    -e "s/\"version\": \"[0-9a-z.-]+\"/\"version\": \"${NEW_VERSION}\"/" \
```
Comment (rhelmer) [inline bin/prepare-release.sh:91]: Off-topic, but something that really parses the JS like `jq` could do this search-and-replace.

File: core-addon/IonCore.js [hunk 2]
```diff
@@ -12,7 +12,7 @@ const ION_DEFAULT_ARGS = {
   // NOTE: if this URL ever changes, you will have to update the domain in
```
Comment (rhelmer) [inline core-addon/IonCore.js:12]: Do you want to rename `IonCore.js` as well or not yet?

Comment (rhelmer) [reply core-addon/IonCore.js:12]: Also `./tests/core-addon/IonCore.test.js`

**Review by rhelmer — APPROVED** [review]
This looks great! The only real question I have is about renaming the `IonCore.js` and related test.

File: bin/prepare-release.sh (hunk 1 above)
Comment (Dexterp37) [reply bin/prepare-release.sh:91]: Good point. Let me file an issue for that!

Comment (Dexterp37) [reply bin/prepare-release.sh:91]: https://github.com/mozilla-rally/core-addon/issues/264

File: core-addon/IonCore.js (hunk 2 above)
Comment (Dexterp37) [reply core-addon/IonCore.js:12]: I'll be dealing with other code changes in #114 if you don't mind, as this PR already got really big :(

**Comment by Dexterp37** [comment]
> https://github.com/mozilla-extensions/ion-core-addon/

There you go! https://github.com/mozilla-rally/core-addon/issues/265

**Pull request merged**